pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -556,80 +556,62 @@ func generateAggregatedProwjobs(uid string, ciopConfig *api.ReleaseBuildConfigur
+		Tests: []api.TestStepConfiguration{
 			{
-				Name: "temp",
-				VolumeSource: corev1.VolumeSource{
-					EmptyDir: &corev1.EmptyDirVolumeSource{},
-				},
-			},
-		},
-		Containers: []corev1.Container{
-			{
-				Image:           "registry.ci.openshift.org/ci/job-run-aggregator",
-				ImagePullPolicy: corev1.PullAlways,
-				Resources: corev1.ResourceRequirements{
-					Requests: corev1.ResourceList{"cpu": *resource.NewMilliQuantity(10, resource.DecimalSI)},
-				},
-				Command: []string{"job-run-aggregator"},
-				Args: []string{
-					"analyze-job-runs",
-					"--google-service-account-credential-file=/secrets/gcs/service-account.json",
-					fmt.Sprintf("--job=%s", jobName),
-					fmt.Sprintf("--aggregation-id=%s", uid),
-					fmt.Sprintf("--explicit-gcs-prefix=logs/%s", submitted),
-					fmt.Sprintf("--job-start-time=%s", startTime.Format(time.RFC3339)),
-					"--working-dir=/tmp",
-				},
-				VolumeMounts: []corev1.VolumeMount{
-					{
-						Name:      "temp",
-						MountPath: "/tmp",
+				As: "release-analysis-prpqr-aggregator",
+				MultiStageTestConfiguration: &api.MultiStageTestConfiguration{
+					Environment: map[string]string{
+						"GOOGLE_SA_CREDENTIAL_FILE": "/var/run/secrets/google-serviceaccount-credentials.json",
+						"VERIFICATION_JOB_NAME":     jobName,
+						"JOB_START_TIME":            startTime.Format(time.RFC3339),
+						"AGGREGATION_ID":            uid,
+						"WORKING_DIR":               "$(ARTIFACT_DIR)/release-analysis-aggregator",
+						"EXPLICIT_GCS_PREFIX":       fmt.Sprintf("logs/%s", submitted),
 					},
-					{
-						Name:      "pull-secret",
-						MountPath: "/etc/pull-secret",
-						ReadOnly:  true,
-					},
-					{
-						Name:      "gcs-credentials",
-						MountPath: api.GCSUploadCredentialsSecretMountPath,
-						ReadOnly:  true,
+					Test: []api.TestStep{
+						{
+							Reference: &[]string{"openshift-release-analysis-prpqr-aggregator"}[0],
+						},
 					},
 				},
 			},
 		},
+		Resources: map[string]api.ResourceRequirements{
+			"*": {
+				Requests: map[string]string{"cpu": "100m", "memory": "200Mi"},
+				Limits:   map[string]string{"memory": "6Gi"},
+			},
+		},
 	}
-	annotations := map[string]string{
-		releaseJobNameAnnotation: jobNameHash(aggregatorJobName),
+
+	unresolvedConfigRaw, err := yaml.Marshal(ciopConfig)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't marshal ci-operator config")
 	}
 
-	periodic := prowconfig.Periodic{JobBase: prowconfig.JobBase{
-		Name:      aggregatorJobName,
-		Namespace: &[]string{prpqrNamespace}[0],
-		Spec:      spec,
-	}}
+	jobBaseGen := prowgen.NewProwJobBaseBuilderForTest(ciopConfig, &prowgen.ProwgenInfo{}, prowgen.NewCiOperatorPodSpecGenerator(), ciopConfig.Tests[0])
+
+	periodic := prowgen.GeneratePeriodicForTest(jobBaseGen, &prowgen.ProwgenInfo{}, "@yearly", "", false, ciopConfig.CanonicalGoRepository)
+	periodic.Name = aggregatorJobName
+
+	periodic.DecorationConfig.Timeout = &prowv1.Duration{Duration: 6 * time.Hour}
```

Comment:
Why 6? Can we describe this somehow?

Reply:
I was advised on how the other aggregator job is configured. See https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/release/openshift-release-infra-periodics.yaml#L2274-L2275. However, the step already has a 6h timeout, but shrug.

Reply:
I am saying if there is no way to configure this, at least it can be explained by giving 6 a proper name so that the next person would know what 6 means actually.

Reply:
Do you mean to add a comment to explain why I put 6hours?

Reply:
Could be a comment but better: a descriptive name for this time period. It's basically a const.

Reply:
But the `periodic.DecorationConfig.Timeout` variable is already descriptive enough.

Reply:
I will try to add a comment to explain why we need to configure a 6-hour timeout.  But to be honest I don't even know why :D. Also, I believe that we could just remove that completely. The step has already a 6-hour timeout. I was hoping that won't affect the whole pod timeout. @bbguimaraes ??

Reply:
If you don't specify a timeout here, the Prow default will kick in and I think that's 2h.

Reply:
@droslean using named constants instead of magic values is a good practice though. You could call `Duration: 6 * time.Hour` something like `defaultAggregatorJobTimeout`, give it a comment like "Aggregator jobs need more time to finish than the jobs they are aggregating (default job timeout is set to 4h in OpenShift CI)" and the result is that whoever comes up here later knows that the value is set intentionally, not as a ballpark number and exactly knows why the number was chosen.

See: https://refactoring.guru/replace-magic-number-with-symbolic-constant

Reply:
addressed